### PR TITLE
front: add User Menu component

### PR DESCRIPTION
**Pull request by lportemo**

### CODE REVIEW

File: front/src/components/UserMenu.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                }}
+            >
+                { profile ? 
+                    ([<MenuItem key={0} onClick={handleClose}>{profile.username}</MenuItem>,
+                    <AdminMenuItems key={1} is_staff={profile.is_staff} />,
+                    <MenuItem key={1000} onClick={toLogout}>Déconnexion</MenuItem>
+                    ])
+                    : (<MenuItem onClick={toLogin}>Connexion</MenuItem>)
+                }
+            </Menu> }
```
Comment (AurelReb) [inline front/src/components/UserMenu.js]: Please use the Dropdown component

File: front/src/components/UserMenu.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+function UserMenu() {
+    const [anchor, setAnchor] = React.useState(false);
+    const [profile, setProfile] = React.useState(null);
+    React.useEffect(() => {
+        if (profile != null) { return; }
+        userApi.getMyProfile()
+            .then(profile => setProfile(profile))
+            .catch(() => setProfile(false));
+    })
```
Comment (AurelReb) [inline front/src/components/UserMenu.js]: You should add this into the `store` so we can access the user's infos everywhere in the app. This way we can show different components depending on the user's status

**Review by AurelReb — CHANGES_REQUESTED** [review]

File: front/src/config/store.js [hunk 3]
```diff
@@ -4,15 +4,18 @@ import produce from 'immer';
...
 const TOGGLE_DARK_THEME = 'TOGGLE_DARK_THEME';
 const FETCH_TRACKS = 'FETCH_TRACKS';
 const FETCH_TRACK = 'FETCH_TRACK';
 const FETCH_PROBLEMS = 'FETCH_PROBLEMS';
 const FETCH_PROBLEM = 'FETCH_PROBLEM';
 const CHANGE_SUBMISSION = 'CHANGE_SUBMISSION';
+const CHANGE_MY_PROFILE = 'CHANGE_MY_PROFILE';
 
 const initialState = {
+  profile: false,
```
Comment (AurelReb) [inline front/src/config/store.js]: is it a boolean?

Comment (lportemo) [reply front/src/config/store.js]: It depends : it is either a Boolean or an object

Comment (lportemo) [reply front/src/config/store.js]: if user is disconnected = false else profile

File: front/src/config/store.js [hunk 4]
```diff
@@ -165,6 +172,18 @@ const fetchSubmissionUntilCorrectionEnds = (submissionId, timeout = 60) => {
...
+  return {type: CHANGE_MY_PROFILE, profile };
+}
+
+const fetchMyProfile = () => {
+  return async (dispatch) => {
+    await userApi.getMyProfile()
+      .catch(() => dispatch(changeMyProfile(false)))
+      .then(profile => dispatch(changeMyProfile(profile)));
+  }
+};
```
Comment (AurelReb) [inline front/src/config/store.js]: `await` allows you not to use .then and .catch methods of a promise.
Please save the response as a `const` and catch the potential errors with a `try{}except{}` statement ;)

File: front/src/config/store.js (hunk 3 above)
Comment (AurelReb) [reply front/src/config/store.js]: hmm it would be better to have either an empty object or a `null`

Comment (lportemo) [reply front/src/config/store.js]: I used null as default and/or disconnected value for profile

File: front/src/config/store.js (hunk 4 above)
Comment (lportemo) [reply front/src/config/store.js]: made a `try{...} catch {...}` block

**Review by AurelReb — APPROVED** [review]

**Pull request merged**